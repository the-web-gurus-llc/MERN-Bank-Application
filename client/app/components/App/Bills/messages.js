import { defineMessages } from 'react-intl';

export const scope = 'app.components.Bills';

export default defineMessages({
  title: {
    id: `${scope}.title`,
    defaultMessage: 'Bills',
  },
  action: {
    id: `${scope}.action`,
    defaultMessage: 'Create a new bill',
  },
  personal: {
    id: `${scope}.personal`,
    defaultMessage: 'Personal',
  },
});
