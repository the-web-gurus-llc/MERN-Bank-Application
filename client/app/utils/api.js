const API_BASE_URL = 'https://api.pietrzakadrian.com/bank';
const AUTH = 'Auth';
const USERS = 'Users';
const CURRENCIES = 'Currencies';
const BILLS = 'Bills';
const TRANSACTIONS = 'Transactions';
const MESSAGES = 'Messages';
const NOTIFICATIONS = 'Notifications';

export default {
  auth: {
    login: `${API_BASE_URL}/${AUTH}/login`,
    register: `${API_BASE_URL}/${AUTH}/register`,
    logout: `${API_BASE_URL}/${AUTH}/logout`,
    forgetPassword: `${API_BASE_URL}/${AUTH}/password/forget`,
    resetPassword: `${API_BASE_URL}/${AUTH}/password/reset`,
  },
  users: (path = '') => {
    switch (path) {
      case 'checkEmail':
        return (email) => `${API_BASE_URL}/${USERS}/${email}/checkEmail`;
      default:
        return `${API_BASE_URL}/${USERS}`;
    }
  },
  currencies: `${API_BASE_URL}/${CURRENCIES}`,
  messages: `${API_BASE_URL}/${MESSAGES}`,
  notifications: `${API_BASE_URL}/${NOTIFICATIONS}`,
  bills: (path = '') => {
    switch (path) {
      case 'amountMoney':
        return `${API_BASE_URL}/${BILLS}/amountMoney`;
      case 'accountBalance':
        return `${API_BASE_URL}/${BILLS}/accountBalance`;
      case 'accountBalanceHistory':
        return `${API_BASE_URL}/${BILLS}/accountBalanceHistory`;
      case 'search':
        return (accountBillNumber) =>
          `${API_BASE_URL}/${BILLS}/${accountBillNumber}/search`;
      default:
        return `${API_BASE_URL}/${BILLS}`;
    }
  },
  transactions: (path = '') => {
    switch (path) {
      case 'create':
        return `${API_BASE_URL}/${TRANSACTIONS}/create`;
      case 'confirm':
        return `${API_BASE_URL}/${TRANSACTIONS}/confirm`;
      case 'authorizationKey':
        return (uuid) =>
          `${API_BASE_URL}/${TRANSACTIONS}/${uuid}/authorizationKey`;
      case 'confirmationFile':
        return (uuid, locale) =>
          `${API_BASE_URL}/${TRANSACTIONS}/${uuid}/${locale}/confirmationFile`;
      default:
        return `${API_BASE_URL}/${TRANSACTIONS}`;
    }
  },
};
