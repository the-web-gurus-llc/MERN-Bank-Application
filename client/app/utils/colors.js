export default {
  white: '#fff',
  black: '#000',
  primaryBlue: '#1890ff',
  grey: 'rgb(242, 244, 247)',
  red: 'rgb(229, 0, 0)',
  redErr: '#ff4d4f',
  silver: '#bababa',
};
