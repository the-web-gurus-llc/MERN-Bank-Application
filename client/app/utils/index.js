export { default as colors } from './colors';
export { default as media } from './media';
export { default as request } from './request';
export { default as typography } from './typography';
export { default as api } from './api';
export { default as routes } from './routes';
export { default as locales } from './locales';
export { default as i18n } from './i18n';
