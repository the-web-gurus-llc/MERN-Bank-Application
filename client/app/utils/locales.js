exports.DEFAULT_LOCALE = 'en';
exports.DEFAULT_DATE_FORMAT = 'dd.MM.yyyy, hh:mm aa';

// prettier-ignore
exports.appLocales = [
  'en',
  'de',
  'pl',
];
