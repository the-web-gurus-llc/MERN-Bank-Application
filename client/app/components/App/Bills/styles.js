import styled from 'styled-components';
import { typography } from 'utils';

export const StyledBillAmountMoney = styled.span`
  font-weight: ${typography.fontWeightBold};
`;
