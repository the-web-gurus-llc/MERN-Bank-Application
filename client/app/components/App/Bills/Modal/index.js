import React from 'react';
import { Button } from 'antd';
import { StyledForm } from 'components/Form/styles';
import { createStructuredSelector } from 'reselect';
import {
  makeSelectIsOpenedModal,
  makeSelectCurrency,
} from 'containers/DashboardPage/selectors';
import { useSelector, useDispatch } from 'react-redux';
import CurrencyToggle from 'components/CurrencyToggle';
import { createNewBillAction } from 'containers/DashboardPage/actions';
import LoadingIndicator from 'components/LoadingIndicator';
import { makeSelectIsLoading } from 'providers/LoadingProvider/selectors';
import { getRequestName } from 'helpers';
import { CREATE_NEW_BILL_REQUEST } from 'containers/DashboardPage/constants';
import { injectIntl, intlShape, FormattedMessage } from 'react-intl';
import { toggleModalAction } from 'containers/App/actions';
import { StyledModal } from 'components/App/Modal/styles';
import messages from './messages';

const stateSelector = createStructuredSelector({
  currency: makeSelectCurrency(),
  isOpenedModal: makeSelectIsOpenedModal(),
  isLoading: makeSelectIsLoading(getRequestName(CREATE_NEW_BILL_REQUEST)),
});

function Modal({ intl }) {
  const [form] = StyledForm.useForm();
  const dispatch = useDispatch();
  const { isOpenedModal, isLoading } = useSelector(stateSelector);
  const snippets = {
    success: {
      title: intl.formatMessage(messages.billHasBeenCreated),
      description: intl.formatMessage(messages.billHasBeenCreatedDescription),
    },
    error: {
      title: intl.formatMessage(messages.billHasNotBeenCreated),
      description: intl.formatMessage(
        messages.billHasNotBeenCreatedDescription,
      ),
    },
  };

  const createNewBill = () => dispatch(createNewBillAction(snippets));
  const toggleModal = () => {
    dispatch(toggleModalAction());
    form.resetFields();
  };

  const onValidateFields = async () => {
    try {
      await form.validateFields();
      createNewBill();
      toggleModal();
      form.resetFields();
    } catch (error) {
      Error(error);
    }
  };

  return (
    <StyledModal
      title={intl.formatMessage(messages.title)}
      visible={isOpenedModal}
      onOk={onValidateFields}
      onCancel={toggleModal}
      centered="true"
      footer={[
        <Button key="back" onClick={toggleModal}>
          <FormattedMessage {...messages.cancel} />
        </Button>,
        <Button
          key="submit"
          disabled={isLoading}
          type="primary"
          onClick={onValidateFields}
        >
          {isLoading ? (
            <LoadingIndicator />
          ) : (
            <FormattedMessage {...messages.create} />
          )}
        </Button>,
      ]}
    >
      <>
        <p>
          <FormattedMessage {...messages.descriptionTop} />
        </p>

        <StyledForm centered="true" form={form} name="create-new-bill">
          <CurrencyToggle tailed="true" />
        </StyledForm>

        <p>
          <FormattedMessage {...messages.descriptionBottom} />
        </p>
      </>
    </StyledModal>
  );
}

Modal.propTypes = {
  intl: intlShape.isRequired,
};

export default injectIntl(Modal);
