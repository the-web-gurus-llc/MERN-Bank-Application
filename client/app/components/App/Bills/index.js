import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { makeSelectBills } from 'containers/DashboardPage/selectors';
import { getBillsAction } from 'containers/DashboardPage/actions';
import { StyledCard } from 'components/App/Card/styles';
import LoadingIndicator from 'components/LoadingIndicator';
import { makeSelectIsLoading } from 'providers/LoadingProvider/selectors';
import { StyledTable, StyledTableWrapper } from 'components/App/Table/styles';
import { PlusCircleOutlined } from '@ant-design/icons';
import { FormattedMessage } from 'react-intl';
import Modal from 'components/App/Bills/Modal';
import { toggleModalAction } from 'containers/App/actions';
import { getRequestName } from 'helpers';
import { GET_BILLS_REQUEST } from 'containers/DashboardPage/constants';
import {
  StyledButton,
  StyledButtonContent,
} from 'components/App/Button/styles';
import { StyledTag } from 'components/App/Tag/styles';
import { StyledBillAmountMoney } from './styles';
import messages from './messages';

const stateSelector = createStructuredSelector({
  bills: makeSelectBills(),
  isLoading: makeSelectIsLoading(getRequestName(GET_BILLS_REQUEST)),
});

const columns = [
  {
    dataIndex: 'accountBillNumber',
    render: (text) => (
      <div>
        {text}{' '}
        <StyledTag color="blue" key={text.uuid}>
          <FormattedMessage {...messages.personal} />
        </StyledTag>
      </div>
    ),
  },
  {
    dataIndex: 'amountMoney',
    render: (text, { currency }) => (
      <>
        <StyledBillAmountMoney>{text}</StyledBillAmountMoney> {currency.name}
      </>
    ),
  },
];

export default function Bills() {
  const dispatch = useDispatch();
  const { bills, isLoading } = useSelector(stateSelector);

  const getBills = () => dispatch(getBillsAction());
  const toggleModal = () => dispatch(toggleModalAction());

  useEffect(() => {
    if (!bills.length) {
      getBills();
    }
  }, []);

  return (
    <>
      <FormattedMessage {...messages.title}>
        {(title) => (
          <StyledCard
            darker="true"
            loaded={isLoading ? 1 : 0}
            bordered={false}
            shadowed="true"
            title={title}
            extra={
              <StyledButton type="link" onClick={toggleModal}>
                <StyledButtonContent onMouseDown={(e) => e.stopPropagation()}>
                  <PlusCircleOutlined />{' '}
                  <FormattedMessage {...messages.action} />
                </StyledButtonContent>
              </StyledButton>
            }
          >
            {isLoading ? (
              <LoadingIndicator />
            ) : (
              <StyledTableWrapper onMouseDown={(e) => e.stopPropagation()}>
                <StyledTable
                  showHeader={false}
                  pagination={false}
                  dataSource={bills}
                  columns={columns}
                  rowKey={({ uuid }) => uuid}
                />
              </StyledTableWrapper>
            )}
          </StyledCard>
        )}
      </FormattedMessage>

      <Modal />
    </>
  );
}
