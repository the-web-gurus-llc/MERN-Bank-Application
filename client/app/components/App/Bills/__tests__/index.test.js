import React from 'react';
import { render } from '@testing-library/react';
import configureStore from 'utils/configureStore';
import { createMemoryHistory } from 'history';
import { Provider } from 'react-redux';
import { IntlProvider } from 'react-intl';
import { DEFAULT_LOCALE } from 'utils/locales';
import Bills from '../index';
import 'utils/__tests__/__mocks__/matchMedia';

describe('<Bills />', () => {
  const history = createMemoryHistory();
  const store = configureStore({}, history);

  it('should render a Bills', () => {
    const { container } = render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <Bills />
        </IntlProvider>
      </Provider>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
