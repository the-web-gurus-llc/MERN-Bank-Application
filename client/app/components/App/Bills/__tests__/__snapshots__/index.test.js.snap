// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Bills /> should render a Bills 1`] = `
<div
  class="ant-card Cardstyle__StyledCard-s807iu-0 heNqZU"
  darker="true"
  loaded="1"
  shadowed="true"
>
  <div
    class="ant-card-head"
  >
    <div
      class="ant-card-head-wrapper"
    >
      <div
        class="ant-card-head-title"
      >
        Bills
      </div>
      <div
        class="ant-card-extra"
      >
        <button
          class="ant-btn Buttonstyle__StyledButton-nh59mr-0 bcqyjQ ant-btn-link"
          type="button"
        >
          <div
            class="Buttonstyle__StyledButtonContent-nh59mr-1 cBxojI"
          >
            <span
              aria-label="plus-circle"
              class="anticon anticon-plus-circle"
              role="img"
            >
              <svg
                aria-hidden="true"
                class=""
                data-icon="plus-circle"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M696 480H544V328c0-4.4-3.6-8-8-8h-48c-4.4 0-8 3.6-8 8v152H328c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8h152v152c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V544h152c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8z"
                />
                <path
                  d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                />
              </svg>
            </span>
             
            <span>
              Create a new bill
            </span>
          </div>
        </button>
      </div>
    </div>
  </div>
  <div
    class="ant-card-body"
  >
    <div
      class="ant-spin ant-spin-spinning LoadingIndicatorstyle__StyledSpin-sc-1hamj2j-0 cVaifN"
    >
      <span
        aria-label="loading"
        class="anticon anticon-loading LoadingIndicatorstyle__StyledLoadingOutlined-sc-1hamj2j-1 bgzkcd ant-spin-dot"
        role="img"
      >
        <svg
          aria-hidden="true"
          class="anticon-spin"
          data-icon="loading"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="0 0 1024 1024"
          width="1em"
        >
          <path
            d="M988 548c-19.9 0-36-16.1-36-36 0-59.4-11.6-117-34.6-171.3a440.45 440.45 0 00-94.3-139.9 437.71 437.71 0 00-139.9-94.3C629 83.6 571.4 72 512 72c-19.9 0-36-16.1-36-36s16.1-36 36-36c69.1 0 136.2 13.5 199.3 40.3C772.3 66 827 103 874 150c47 47 83.9 101.8 109.7 162.7 26.7 63.1 40.2 130.2 40.2 199.3.1 19.9-16 36-35.9 36z"
          />
        </svg>
      </span>
    </div>
  </div>
</div>
`;
